src/main/java/jcifs/smb1/smb1/SmbSession.java

```diff
@@ -0,0 +1,476 @@
+                                byte[] signingKey = auth.getSigningKey(transport.server.encryptionKey);
+                                request.digest = new SigningDigest(signingKey, false);
+                            }
+                        }
+    
+                        request.auth = auth;
+    
+                        try {
+                            transport.send( request, response );
+                        } catch (SmbAuthException sae) {
+                            throw sae;
+                        } catch (SmbException se) {
+                            ex = se;
+                        }
+    
+                        if( response.isLoggedInAsGuest &&
+                                    "GUEST".equalsIgnoreCase( auth.username ) == false &&
+                                    transport.server.security != SmbConstants.SECURITY_SHARE &&
+                                    auth != NtlmPasswordAuthentication.ANONYMOUS) {
+                            throw new SmbAuthException( NtStatus.NT_STATUS_LOGON_FAILURE );
+                        }
+    
+                        if (ex != null)
+                            throw ex;
+    
+                        uid = response.uid;
+    
+                        if( request.digest != null ) {
+                            /* success - install the signing digest */
+                            transport.digest = request.digest;
+                        }
+    
+                        connectionState = 2;    
+
+                        state = 0;
+    
+                        break;
+                    case 20:
+                        if (nctx == null) {
+                            boolean doSigning = (transport.flags2 & ServerMessageBlock.FLAGS2_SECURITY_SIGNATURES) != 0;
+                            nctx = new NtlmContext(auth, doSigning);
+                        }
+    
+                        if (SmbTransport.log.level >= 4)
+                            SmbTransport.log.println(nctx);
+    
+                        if (nctx.isEstablished()) {
+
+                            netbiosName = nctx.getNetbiosName();
+
+                            connectionState = 2;
+
+                            state = 0;
+                            break;
+                        }
+    
+                        try {
+                            token = nctx.initSecContext(token, 0, token.length);
+                        } catch (SmbException se) {
+                            /* We must close the transport or the server will be expecting a
+                             * Type3Message. Otherwise, when we send a Type1Message it will return
+                             * "Invalid parameter".
+                             */
+                            try { transport.disconnect(true); } catch (IOException ioe) {}
+                            uid = 0;
+                            throw se;
+                        }
+    
+                        if (token != null) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of != on line 378 to == check before setting the field value on line 421 or verify if the check is needed at all.